fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java

```diff
@@ -152,6 +170,86 @@ public void processTaskRun(TaskRunContext context) throws Exception {
 
         // insert execute successfully, update the meta of materialized view according to ExecPlan
         updateMeta(execPlan);
+
+        if (mvContext.hasNextBatchPartition()) {
+            generateNextTaskRun();
+        }
+    }
+
+    @VisibleForTesting
+    public void filterPartitionByRefreshNumber(Set<String> partitionsToRefresh, MaterializedView materializedView) {
```

Comment:
have you test if one source base table partition has two target mv partitions can run successfully?

Reply:
In fact, this configuration has nothing to do with the source table, only the number of partitions of the materialized view table.